sensu/plugins/check-neutron-agents-duplicate.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+#
+# Calls neutron agent api, checks for duplicate with same hostname
+#
+# return CRITICAL if any duplicate agent on same host found
+#
+# Jose L Coello Enriquez <jlcoello@us.ibm.com>
+
+import argparse
+import sys
+import os
+import json
+import subprocess
+import re
+import requests
+
+
+STATE_OK = 0
+STATE_WARNING = 1
+STATE_CRITICAL = 2
+
+timeout = 30
+
+def run(command, stdin=None):
+    if stdin:
+        p = subprocess.Popen(command, shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
+        stdout, stderr = p.communicate(stdin)
+    else:
+        p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
+        stdout, stderr = p.communicate()
+
+    stdout = stdout.strip()
+    if p.returncode == 0 and stdout:
+        return stdout
+    else:
+        print("WARN: \"%s\" returned %s: %s" % (command, p.returncode, stderr))
+        sys.exit(STATE_WARNING)
+
+def request_token(url, method='GET', retries=2, **kwargs):
```

Comment:
It looks like request_token() and request() are almost identical except a minor difference in the last return statement and we could probably do with just one of these functions?